packages/extensions/README.md

```diff
@@ -0,0 +1,180 @@
+
+4. **Modify the app pages** to use components from the `diracx-components` library (e.g., providers, apps).
+
+### Architecture
+
+We strongly recommend following the directory structure below to keep your project organized:
+
+- `/src/app`: Contains the main application logic and Next.js setup. This directory houses the core of the application built using Next.js, where each page.tsx file represents a page in the application with [folder-based routing](https://nextjs.org/docs/app/building-your-application/routing). [Next.js Official Documentation](https://nextjs.org/docs)
+
+- `/src/<your extension>`: This directory includes the source code related to your extension. You can create custom components, hooks, ... in this directory.
+  - `/src/<your extension>/components`: Contains custom React components. This folder includes reusable UI components built using React. Components in React are independent, reusable pieces of UI that can manage their own state. [React Components](https://reactjs.org/docs/components-and-props.html)
+  - `/src/<your extension>/contexts`: Manages global state using React Context. This folder contains context providers which are used to manage and share global state across the application. React Context provides a way to pass data through the component tree without having to pass props down manually at every level. [React Context](https://reactjs.org/docs/context.html)
+  - `/src/<your extension>/hooks`: Custom React hooks for encapsulating reusable logic. This directory includes custom hooks. Hooks are special functions that let you "hook into" React state and lifecycle features from function components. [React Hooks](https://reactjs.org/docs/hooks-intro.html)
+  - `/src/<your extension>/types`: TypeScript type definitions for the application. This folder contains TypeScript type definitions to ensure type safety throughout the application.
+
+### Running the Extension with DiracX Charts
+
+To start your DiracX extension follow these steps:
+
+1. **Clone the `diracx-charts` repository** in a parent directory:
+
+   ```bash
+   git clone git@github.com:DIRACGrid/diracx-charts.git
+   ```
+
+2. **Run the demo script** with the path to your extension:
+
+   ```bash
+   ./diracx-charts/run_demo.sh path/to/your-extension
+   ```
+
+This will run the DiracX Demo with your extension.
+
+## Customizing the Extension
+
+You can customize your extension by modifying the files in the `src` directory. This is where you’ll find the main components and logic of your extension.
+
+Having a directory dedicated to your extension components will help you keep your code organized and easy to maintain.
+
+### Extending the DiracX Apps
+
+To add new apps to your extension, you can create new components in your extension directory.
+
+[`testApp`](src/gubbins/components/TestApp/testApp.tsx) provides an example of a basic app component.
```

Comment:
We should probably mention that `diracx-web-components` can be visually observed in the storybook documentation, which would probably be located at https://diracgrid.github.io/diracx-web/